docs/support-matrix.md

```diff
@@ -0,0 +1,43 @@
+# Supported Storage Providers
+
+Ark supports sending backup data to a variety of object storage providers, as well as volume snapshots using either a plugin for the volume provider, or [restic][4] for filesystem level backups.
+
+## In-tree providers
```

Comment:
I'm thinking a single matrix that includes in-tree and plugins would be better. Something that looks like https://github.com/kubernetes/client-go/#compatibility-matrix. We may need 2 matrices: 1 for object storage, and 1 for volumes.

Reply:
I want to make sure we're clear on what the Ark team will take responsibility for. Encoding too much information into columns can make the tabled width get out of hand, which is what I'd like to avoid.

I'll give it a shot and see how it looks, though.

Reply:
@ncdc Clarifying your suggestion - are you thinking of column headers being the versions of Ark?

Reply:
Brainstorming (data below not necessarily accurate):

Provider|Backup Storage|Volume Snapshots|Docs|Owner|Compatibility
-|-|-|-|-|-
AWS|✅|✅|[Setup Guide](todo)|Ark Team|All Ark versions
Azure|✅|✅ <sup>1</sup>|[Setup Guide](todo)|Ark Team|All Ark versions
Portworx|❌|✅ |[Setup Guide](https://docs.portworx.com/scheduler/kubernetes/ark.html)|PortWorx|Ark 0.8+

<sup>1</sup> Only Azure managed disks are supported